pkg/pgmodel/extension.go

```diff
@@ -76,62 +76,70 @@ func checkPromscaleExtensionVersion(conn *pgx.Conn) error {
 	return nil
 }
 
-func migrateExtension(conn *pgx.Conn) error {
-	availableVersions, err := fetchAvailableExtensionVersions(conn)
-	ExtensionIsInstalled = false
+func migrateExtension(conn *pgx.Conn, extName string, extSchemaName string, validRange semver.Range, rangeString string) error {
+	availableVersions, err := fetchAvailableExtensionVersions(conn, extName)
 	if err != nil {
 		return err
 	}
 	if len(availableVersions) == 0 {
-		log.Warn("msg", "the promscale extension is not available, proceeding without extension")
-		return nil
+		return fmt.Errorf("the extension is not available")
 	}
 
-	newVersion, ok := getNewExtensionVersion(availableVersions)
+	newVersion, ok := getNewExtensionVersion(availableVersions, validRange)
 	if !ok {
-		log.Warn("msg", "the promscale extension is not available at the right version, need version: %v, proceeding without extension", version.ExtVersionRangeString)
-		return nil
+		return fmt.Errorf("the extension is not available at the right version, need version: %v", rangeString)
 	}
 
-	currentVersion, isInstalled, err := fetchInstalledExtensionVersion(conn, "promscale")
+	currentVersion, isInstalled, err := fetchInstalledExtensionVersion(conn, extName)
 	if err != nil {
 		return fmt.Errorf("Could not get the installed extension version: %w", err)
 	}
 
 	if !isInstalled {
 		_, extErr := conn.Exec(context.Background(),
-			fmt.Sprintf("CREATE EXTENSION IF NOT EXISTS promscale WITH SCHEMA %s VERSION '%s'",
-				extSchema, getSqlVersion(newVersion)))
+			fmt.Sprintf("CREATE EXTENSION IF NOT EXISTS %s WITH SCHEMA %s VERSION '%s'",
+				extName, extSchemaName, getSqlVersion(newVersion, extName)))
 		if extErr != nil {
 			return extErr
 		}
-		return checkExtensionsVersion(conn)
+		return nil
 	}
 
 	comparator := currentVersion.Compare(newVersion)
 	if comparator > 0 {
 		//currentVersion greater than what we can handle, don't use the extension
-		log.Warn("msg", "the promscale extension at a greater version than supported by the connector: %v > %v, proceeding without extension", currentVersion, newVersion)
+		return fmt.Errorf("the extension at a greater version than supported by the connector: %v > %v", currentVersion, newVersion)
```

Comment:
Why did we not specify promscale extension here?

Reply:
because this now runs for both extensions. The returned errors are wrapped in parent errors that specify which extension

Reply:
Oh, then its fine.